### ISSUE TITLE
segment criteria - RFM

### ISSUE THREAD

**Issue by SSMK-wq** [sonwanesuresh95/rfm#1]
Hi,

I recently came across this package and it is very useful package. Helps people who lack extensive coding knowledge. Appreciate it.

However, I have one quick question. 

How can I find out the definition and RFM score criteria for each segment?

For ex: I get a graph like below and how do I customers with which score go into Champions segment?

Can direct me to a documentation where I can find the definition and score criteria for each segment? There is no documentation link in your readme. page

Meaning, if it is 555 it is champions. What if it is 541? is it still considered champions? So, I would like to know the cut off points that you chose for each segment.

`How much drop in revenue value or recency value or frequency value is required for a record to be put into another segment? Is it 100$ or 2 days for recency or 2 points for frequency? 

![image](https://user-images.githubusercontent.com/30723825/165887079-aa2c1a40-812b-4e45-959a-1196162be9ac.png)

Basically, am looking for a table like below which I got from tutorial [https://www.analyticseducator.com/Blog/Customer_Segmentation_Using_RFM_with_Python.html](url)

![image](https://user-images.githubusercontent.com/30723825/165887631-336ef28c-d0cf-4058-aae1-e7e0063f212e.png)

Additionally, is this a new package? Can it be used for real world business insights or is it still under testing?

Nonetheless, great effort and useful package for people like me

**Comment by sonwanesuresh95** [comment]
Thanks for pointing out a few things.
You can make use of this package in real world projects/ business problems. Just make sure your dependencies are installed.

The 10 categories cutoffs are given here:
<table>
<tr>
         <td>Sr. No</td><td>segment_names            </td><td>Recency_seg           </td><td>Freq_seg           </td><td>Monetary_seg</td>
</tr><tr>
  <td>1 </td><td>    Champions    </td><td>                        [4,5]       </td><td>              [4,5]      </td><td>               [4,5]</td>
</tr><tr>
 <td> 2  </td><td>   Loyal Accounts         </td><td>             [3,5]      </td><td>               [3,5]    </td><td>                 [3,5]</td>
</tr><tr>
  <td>3 </td><td>    Potential Loyalist   </td><td>                [3,5]    </td><td>                 [2,3]   </td><td>                  [2,4]</td>
</tr><tr>
  <td>4 </td><td>    Promising          </td><td>                    [4,5]  </td><td>                   [1,2]   </td><td>                  [3,5]</td>
</tr><tr>
  <td>5  </td><td>  New Active Accounts   </td><td>          [5,5]  </td><td>                   [1,1]   </td><td>                  [1,5]</td>
</tr><tr>
  <td>6 </td><td>   Need Attention </td><td>                     [2,3]     </td><td>                [1,2]    </td><td>                 [4,5]</td>
</tr><tr>
  <td>7 </td><td>   About To Sleep  </td><td>                    [2,3]   </td><td>                  [1,2]   </td><td>                  [1,3]</td>
</tr><tr>
  <td>8 </td><td>   At Risk             </td><td>                      [1,2]   </td><td>                  [1,5]     </td><td>                [3,5]</td>
</tr><tr>
  <td>9 </td><td>   Lost              </td><td>                         [1,2]      </td><td>               [1,5]     </td><td>                [1,2]</td>
</tr><tr>
  <td>10 </td><td> Low spenders   </td><td>                     [3,5]     </td><td>                [1,5]      </td><td>               [1,2]</td>
</tr>
</table>

Me and my team have came up with these cutoffs. they are tested out and well suited to any sort of data.
I have written a blog regarding the same topic. Find it here: https://medium.com/@suresh-sonwane/rfm-analysis-python-customer-segmentation-182a2bbcd351
For any questions, please reach out ot me: sonwanesuresh95@gmail.com


I have been working on some additional functions like dynamic binning, to counter the problem of skewed recency, frequency and monetary columns. If these columns are skewed, binning is done inaccurately. (based on quartiles)
Although for now, this functionality is in beta mode. In next release it will be finished and documented.
